src/kanata/linux.rs

```diff
@@ -113,3 +122,68 @@ impl Kanata {
         Ok(())
     }
 }
+
+/// Returns true if the scroll event should be sent to the processing loop, otherwise returns
+/// false.
+fn handle_scroll(
+    kanata: &Mutex<Kanata>,
+    in_event: InputEvent,
+    code: OsCode,
+    all_events: &[InputEvent],
+) -> Result<bool> {
+    let direction: MWheelDirection = code.try_into().unwrap();
+    let scroll_distance = in_event.value().unsigned_abs() as u16;
+    match in_event.kind() {
+        InputEventKind::RelAxis(axis_type) => {
+            match axis_type {
+                RelativeAxisType::REL_WHEEL | RelativeAxisType::REL_HWHEEL => {
+                    if MAPPED_KEYS.lock().contains(&code) {
+                        return Ok(true);
+                    }
+                    // If we just used `write_raw` here, some of the scrolls issued by kanata would be
+                    // REL_WHEEL_HI_RES + REL_WHEEL and some just REL_WHEEL and an issue like this one
+                    // would happen: https://github.com/jtroo/kanata/issues/395
+                    //
+                    // So to fix this case, we need to use `scroll` which will also send hi-res scrolls
+                    // along normal scrolls.
+                    //
+                    // However, if this is a normal scroll event, it may be sent alongside a hi-res
+                    // scroll event. In this scenario, the hi-res event should be used to call
+                    // scroll, and not the normal event. Otherwise, too much scrolling will happen.
```

Comment:
Hm I'm seeing an unusual behaviour where when I test hi-res scrolling purely via kanata - since I don't have a hi-res mouse, gnome-terminal has behaviour where scrolling seems to be delayed for a moment, maybe until the first non-hi-res event is sent?

I wonder if this is a quirk with the app or the DE though, where some inputs are ignored for a while, like noise suppression or touchpad-ignore behaviour. If this code works properly when forwarding inputs from a hi-res mouse though, then should be fine as-is.

Reply:
Ok, I definitely think it's until the joint hi-res+standard event is sent, when testing with `(mwheel-up 50 2)` its very obvious. Would be curious to see if kanata needs to change its `scroll` behaviour when comparing against a hi-res mouse. Maybe hi-res mice send a standard event on the very first hi-res event after a pause, and after that only when accumulating sufficient hi-res scrolling?

Reply:
> Would be curious to see if kanata needs to change its `scroll` behaviour when comparing against a hi-res mouse.

Can't help there. My mouse has a setting to enable hi-res scrolling, but for some reason switching to hires mode doesn't work, it just ignores the setting and stays on normal mode, even on Windows.

Reply:
Reading through some online discussions, it seems hi-res scrolling can sometimes overly sensitive and cause unwanted jitters. Maybe the weird behaviour is intentional as a mitigation for this issue. Well, if the code doesn't work for someone then hopefully will get a bug report.